modules/common/log/logs-process.nix

```diff
@@ -0,0 +1,78 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  cfg = config.ghaf.logging.client;
+  listenerAddress = "admin-vm-debug";
+  listenerPort = config.ghaf.logging.listener.port;
+  # TODO: replace sshCommand and MacCommand with givc rpc to retrieve Mac Address
+  sshCommand = "${pkgs.sshpass}/bin/sshpass -p ghaf ${pkgs.openssh}/bin/ssh -o StrictHostKeyChecking=no ghaf@net-vm";
+  MacCommand = "cat /sys/class/net/wlp0s5f0/address";
+  remoteLoggingAddress = config.ghaf.logging.server.address;
+  loggingPort = config.ghaf.logging.port;
+  lokiForward =
+    if config.ghaf.logging.server.enable
+    then "loki.write.local.receiver, loki.write.remote.receiver"
+    else "loki.write.local.receiver";
+in {
+  config = lib.mkIf cfg.enable {
+    # TODO: Remove hw-mac.service and replace with givc rpc later
+    systemd.services."hw-mac" = {
+      description = "Retrieve MAC address from net-vm";
+      wantedBy = ["alloy.service"];
+      requires = ["network-online.target"];
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = "yes";
+        # Make sure we can ssh before we retrieve mac address
+        ExecStartPre = "${sshCommand} ls";
+        ExecStart = ''
+          ${pkgs.bash}/bin/bash -c "echo -n $(${sshCommand} ${MacCommand}) > /var/lib/private/alloy/MACAddress"
+        '';
+      };
+    };
+
+    environment.etc."alloy/config.alloy" = {
+      text = ''
+        local.file "macAddress" {
+          // Alloy service can read file in this specific location
+          filename = "/var/lib/private/alloy/MACAddress"
+        }
+
+        loki.write "local" {
+          endpoint {
+            url = "http://localhost:${toString loggingPort}/loki/api/v1/push"
+          }
+          external_labels = {systemdJournalLogs = local.file.macAddress.content }
+        }
+
+        loki.write "remote" {
+          endpoint {
+            url = "http://${remoteLoggingAddress}:${toString loggingPort}/loki/api/v1/push"
```

Comment:
I'd just ask for the full endpoint in the module system (including path) here. We might want to change the scheme, or path coming after that. Just asking for an URL is gonna be less work later.

Reply:
Done.

Reply:
No, this is still not a config option, but some math in a let binding. See the other comments.

Reply:
Fixed.